bndtools.api/src/org/bndtools/api/central/ICentral.java

```diff
@@ -0,0 +1,22 @@
+package org.bndtools.api.central;
+
+import org.bndtools.api.ModelListener;
+import org.eclipse.jdt.core.IJavaProject;
+import org.osgi.annotation.versioning.ProviderType;
+
+import aQute.bnd.build.Project;
+
+@ProviderType
+public interface ICentral {
+
+	Project getModel(IJavaProject project);
```

Comment:
```suggestion
	Optional<Project> getModel(IProject project);
```

Reply:
Our codebase hasn't made heavy use of the `Optional` pattern. But I suppose we probably should...

Reply:
Just noticed that this was also a change of the argument type from `IJavaProject` to `IProject`. I'm curious as to the reasoning? Every Bnd project is also a Java project.

Reply:
It is easy to get from `IJavaProject` > `IProject` but the other is not that easy (and better handled on a central 😛place), also as it is maybe now the case who knows what the future brings?
So usually  `IProject` is much more comfortable to use from api user point of view as you almost ever have one at hand.

Reply:
> Our codebase hasn't made heavy use of the Optional pattern. But I suppose we probably should...

I've been using Result more and more. This has all the methods of Optional but

* it allows to carry a message when it fails,
* easy message construction with improved String.format,
* It allows lambdas that throw exceptions.

The idea is derived from Rust.